### ISSUE TITLE
[Bug] Blank config lines sometimes treated as continuations [sf#25]

### ISSUE THREAD

**Issue by bebehei** [rsnapshot/sourceforge-issues#37]
**Reported by hashproduct on 2010-06-03 15:44 UTC**
As described here, the configuration file parser incorrectly treats a blank line as a continuation of the previous line in some cases:
https://sourceforge.net/mailarchive/message.php?msg\_name=20100602231854.GZ5272%40boombox.keegel.wattle.id.au

**Comment by bebehei** [comment]
**Commented by trevorhobson on 2010-10-04 14:30 UTC**
This patch will fix the problem:

```
\--- rsnapshot-program.pl
+++ rsnapshot-program.pl
@@ -529,8 +529,8 @@
\# ignore blank lines
if \(is\_blank\($line\)\) \{ next; \}

\-        \# if the next line begins with space or tab it belongs to this line
\-        while \(defined \($configs\[$file\_line\_num\]\) && $configs\[$file\_line\_num\] =~ /^\(\t|\s\)/\) \{
\+        \# if the next line begins with space or tab and is not blank it belongs to this line
\+        while \(defined\($configs\[$file\_line\_num\]\) && $configs\[$file\_line\_num\] =~ /^\(\t|\s\)/ && \!is\_blank\($configs\[$file\_line\_num\]\)\) \{
\(my $newline = $configs\[$file\_line\_num\]\) =~ s/^\s+|\s+$//g;
$line = $line . "\t" . $newline;
$file\_line\_num++;
```

**Comment by bebehei** [comment]
As I look into the code, this does not apply anymore to rsnapshot. The reported bug is already fixed at rsnapshot/rsnapshot@df5f150251f4d71e766202a24921ffd4ab3198ac
